README.md

```diff
@@ -1 +1,460 @@
-# golangcodestyle
\ No newline at end of file
+# Backend (Golang) Coding Guidelines
```

Comment:
There are few things I would like to add here. 
1. No `TestMain` unless absolutely necessary. The reasoning behind it should be put in the function comment. 
2. Do not initialise structs with empty values. It's not necessary and does not improve readability. 
**Bad**:
```Go
type MyStruct{
ValueA int 
ValueB string 
ValueC []float
}

var empty = MyStruct{
  ValueA = 0,
ValueB = "",
ValueC= nil
}
```

**Good**
```Go
type MyStruct{
ValueA int 
ValueB string 
ValueC []float
}

var empty = MyStruct{}
```

Reply:
added: 
- https://github.com/apaliavy/golangcodestyle/blob/1eda2a9506765ca11274f4bf973baf16a84174af/README.md#main
- https://github.com/apaliavy/golangcodestyle/blob/1eda2a9506765ca11274f4bf973baf16a84174af/README.md#structures-initialization